src/states/node.rs

```diff
@@ -2298,18 +2338,19 @@ impl Node {
                 Some(their_sections) => their_sections,
             };
             let mut version = 1;
-            let mut our_merged_section = BTreeSet::new();
-            for (ver_pfx, peers) in our_sections.iter().chain(&their_sections) {
+            for (ver_pfx, _peers) in our_sections.iter().chain(&their_sections) {
                 if ver_pfx.prefix().is_extension_of(&merge_prefix) {
                     version = cmp::max(version, ver_pfx.version() + 1);
-                    our_merged_section.extend(peers.into_iter().map(|peer| *peer.name()));
                 }
             }
+            let (_, merged_section) = unwrap!(self.our_merged_section_cache.remove(&merge_prefix));
             self.process_own_section_merge(their_prefix,
                                            version,
                                            their_sections,
-                                           our_merged_section,
+                                           merged_section.clone(),
                                            outbox);
+            let _ = self.our_merged_section_cache
+                .insert(merge_prefix, (true, merged_section));
```

Comment:
Why remove and insert it instead of `get_mut` and setting the first component to `true`?